services/autoscaling/tests/unit/conftest.py

```diff
@@ -378,11 +378,17 @@ def enabled_rabbitmq(
     return rabbit_service
 
 
+_LIFESPAN_TIMEOUT: Final[int] = 10
```

Comment:
:+1:

Reply:
Actually, i think the fact that it does not call `on_shutdown` when `on_startup` raises might be solved by using the new lifespan event  https://fastapi.tiangolo.com/advanced/events/#lifespan-events 
Note that `on_startup` and `on_shtudonw` is deprecated https://fastapi.tiangolo.com/advanced/events/#alternative-events-deprecated

Reply:
yep once we migrate to fastapi 0.1xx